### PR TITLE
New methods for registering a texture

### PR DESCRIPTION
**Pull request by SimonWang9610**
## Description & motivation

Fix: https://github.com/shogo4405/HaishinKit.dart/issues/45

Split the logic of `registerTexture` into 3 different methods:
1. `registerTexture()` to create `NetStreamDrawableTexture`
2. `updateTextureSize()` to set the media size for the created `NetStreamDrawableTexture`
3. `unregisterTexture()` to unregister the created texture

For the change on the Android side,
1. `registureTexture()` just removes the logic of updating the media size for the created texture into `updateTextureSize()`
2. `updateTextureSize()` would update the media size of `NetStreamDrawableTexture` if `NetStream.drawable != null`
3. `unregisterTexture()` does nothing and returns null directly, as there is no documentation and interface for us to actually unregister a texture.

For the change on the iOS side:
1. `registerTexture()` would create a `NetStreamDrawableTexture` and store its id as `self.textureId`. By doing so, we ensure multiple invocations on `registerTexture()` just create `NetStreamDrawableTexture` once.
2. `updateTextureSize()`, same as the Android side
3. `unregisterTexture()` would unregister the created texture. It is required by the [FlutterTextureRegistry](https://api.flutter.dev/objcdoc/Protocols/FlutterTextureRegistry.html#/c:objc(pl)FlutterTextureRegistry(im)unregisterTexture:)

Other changes:
A new `Example2` is provided in the `example/` project.

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -9,10 +9,25 @@ import 'package:haishin_kit/rtmp_connection.dart';
...
 import 'package:haishin_kit/video_source.dart';
+import 'package:haishin_kit_example/camera_live_controller.dart';
+import 'package:haishin_kit_example/models/live_stream_state.dart';
 import 'package:permission_handler/permission_handler.dart';
 
-void main() {
-  runApp(const MyApp());
+void main() async {
+  const String server = "rtmp://192.168.68.52/live";
+  const streamKey = "SySD6EQb6";
```
Comment (shogo4405) [inline example/lib/main.dart:18]: [nits] I think it's a test environment, so I'll make changes after merging.

File: example/lib/main.dart [hunk 2]
```diff
@@ -56,10 +71,11 @@ class _MyAppState extends State<MyApp> {
 
     RtmpConnection connection = await RtmpConnection.create();
     connection.eventChannel.receiveBroadcastStream().listen((event) {
+      print("event: $event");
       switch (event["data"]["code"]) {
         case 'NetConnection.Connect.Success':
           if (_mode == "publish") {
-            _stream?.publish("live");
+            _stream?.publish("SySD6EQb6");
```
Comment (shogo4405) [inline example/lib/main.dart:78]: [nits] I think it's a test environment, so I'll make changes after merging.

**Review by shogo4405 — APPROVED** [review]
👍

**Comment by shogo4405** [comment]
Thanks. Do you need to `dart pub publish` ASAP?

**Comment by SimonWang9610** [comment]
> Thanks. Do you need to `dart pub publish` ASAP?

Not urgent, as I am working on diagnosing some issues about `RtmpConnection.dispose/close` and `RtmpStream.dispose/close` at the dart side. 

Incorrect usage of `RtmpConnection.dispose/close` and `RtmpStream.dispose/close` may crash Flutter applications, so I still need some time to diagnose and try to fix them.

I think it would be a good time to publish a new version once they are fixed or well-documented how to use those methods.

what do you think about that?

**Comment by shogo4405** [comment]
If SimonWang9610 is not in a hurry, I have no objections.

I've designed it with the premise that the close() method is intended for disconnecting, and the Stream instance remains valid for continued connection.
The dispose() method is designed for memory release, and after that, the same Stream instance cannot be used.

**Pull request merged**